agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/exporter/Exporter.java

```diff
@@ -828,12 +878,17 @@ private static String nullAwareConcat(String str1, String str2, String separator
   private void exportEvents(
       SpanData span, @Nullable String operationName, float samplingPercentage) {
     for (EventData event : span.getEvents()) {
-      boolean lettuce51 =
-          span.getInstrumentationLibraryInfo().getName().equals("io.opentelemetry.lettuce-5.1");
+      String instrumentationLibraryName = span.getInstrumentationLibraryInfo().getName();
+      boolean lettuce51 = instrumentationLibraryName.equals("io.opentelemetry.lettuce-5.1");
       if (lettuce51 && event.getName().startsWith("redis.encode.")) {
         // special case as these are noisy and come from the underlying library itself
         continue;
       }
+      boolean grpc16 = instrumentationLibraryName.equals("io.opentelemetry.grpc-1.6");
```

Comment:
for special case, can you add a comment linked to the PR or issue from otel? it will offer some context.